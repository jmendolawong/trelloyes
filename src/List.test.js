import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';


describe('List Component', ()=>{
    it('renders without crashing', ()=>{
        const div = document.createElement('div');
        ReactDOM.render(<List title="title test" cards={[{id:'1',title:'test',content:'test'},{id:'2',title:'test',content:'test'}]}/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', ()=>{
        const tree = renderer
          .create(<List title="title test" cards={[{id:'1',title:'test',content:'test'},{id:'2',title:'test',content:'test'}]} />)
          .toJSON();
        expect(tree).toMatchSnapshot();
    });

});