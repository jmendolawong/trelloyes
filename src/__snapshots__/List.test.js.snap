// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component renders the UI as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3>
    title test
  </h3>
  <p>
    content test
  </p>
</div>
`;

exports[`List Component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2 />
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        test
      </h3>
      <p>
        test
      </p>
    </div>
    <div
      className="Card"
    >
      <button
        type="button"
      >
        delete
      </button>
      <h3>
        test
      </h3>
      <p>
        test
      </p>
    </div>
    <button
      className="List-add-button"
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
